src/Rule/Each.php

```diff
@@ -37,20 +37,27 @@ protected function validateValue($value, ValidationContext $context = null): Res
             return $result;
         }
 
-        foreach ($value as $item) {
+        foreach ($value as $index => $item) {
             $itemResult = $this->rules->validate($item, $context);
-            if ($itemResult->isValid() === false) {
-                foreach ($itemResult->getErrors() as $error) {
-                    $result->addError(
-                        $this->formatMessage(
-                            $this->message,
-                            [
-                                'error' => $error,
-                                'value' => $item,
-                            ]
-                        )
-                    );
+            if ($itemResult->isValid()) {
+                continue;
+            }
+
+            foreach ($itemResult->getErrorObjects() as $error) {
+                if (!is_array($item)) {
+                    $errorKey = [$index];
+                    $formatMessage = true;
+                } else {
+                    $errorKey = array_merge([$index], $error->getValuePath());
```

Comment:
```suggestion
                    $errorKey = [$index, ...$error->getValuePath()];
```

Reply:
:heavy_check_mark:

Reply:
Reverted because of Psalm error. https://github.com/yiisoft/validator/runs/5076273960?check_suite_focus=true What should we do about this?

Reply:
Modify `Error` for fix psalm error:

```php
final class Error
{
    private string $message;

    /**
     * @psalm-var list<int|string>
     */
    private array $valuePath;

    /**
     * @psalm-param list<int|string> $valuePath
     */
    public function __construct(string $message, array $valuePath = [])
    {
        $this->message = $message;
        $this->valuePath = $valuePath;
    }

    public function getMessage(): string
    {
        return $this->message;
    }

    /**
     * @psalm-return list<int|string>
     */
    public function getValuePath(): array
    {
        return $this->valuePath;
    }
}
```

Reply:
:heavy_check_mark: